doc/languages-frameworks/vim.section.md

```diff
@@ -154,6 +154,37 @@ in
 }
 ```
 
+### Specificities for some plugins
+#### Tree sitter
+
+By default `nvim-treesitter` encourages you to download, compile and install
+the required tree-sitter grammars at run time with `:TSInstall`. This works
+poorly on NixOS.  Instead, to install the `nvim-treesitter` plugins with a set
+of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
+
+```nix
+{ pkgs, ... }: {
+  environment.systemPackages = [
+    (pkgs.neovim.override {
+      configure = {
+        packages.myPlugins = with pkgs.vimPlugins; {
```

Comment:
Probably worth mentioning: to get a working (Nix) syntax highlighter you also need this:
```
customRC = ''
  lua <<EOF
  -- Enable treesitter syntax highlighting and indentation 
  require"nvim-treesitter.configs".setup{
    highlight={enable=true},
    indent={enable=true},
  }
  -- Manually set the filetype (needed if vim-nix is not installed)
  vim.api.nvim_command("autocmd BufNewFile,BufRead *.nix setlocal filetype=nix")
 EOF
'';
```

Reply:
Given that tree-sitter is to be considered experimental until neovim 0.6, let's not recommend it. `vim-nix` is fine.